### PR TITLE
Add It_Framework PreCommit

### PR DESCRIPTION
**Pull request by Abacn**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Abacn** [comment]
`./gradlew  -p it build` runs successfully in local:

image

**Comment by pranavbhandari24** [comment]
Nice! LGTM
Thanks Yi

**Review by pranavbhandari24 — APPROVED** [review]

File: .github/workflows/beam_PreCommit_ItFramework.yml [hunk 1]
```diff
@@ -83,6 +83,13 @@ jobs:
         requires-py-38: false
         requires-py-39: false
         requires-go: false
+    - name: Authenticate on GCP
```
Comment (pranavbhandari24) [inline .github/workflows/beam_PreCommit_ItFramework.yml:86]: Are these needed by the tests? If I am not wrong, the credentials are only needed for IT's/LT's, which aren't run in this PreCommit?

Comment (Abacn) [reply .github/workflows/beam_PreCommit_ItFramework.yml:86]: Yes, it is needed: it appears resource managers still needs credential setup

Failed run (without credential): https://github.com/Abacn/beam/actions/runs/5682012376/job/15399424990

successful run (with Authenticate on GCP step):

https://github.com/Abacn/beam/actions/runs/5682497039/job/15401009175

Comment (pranavbhandari24) [reply .github/workflows/beam_PreCommit_ItFramework.yml:86]: Ah, gotcha

**Pull request merged**